src/test/msixtest/api_manifestreader.cpp

```diff
@@ -215,6 +212,52 @@ TEST_CASE("Api_AppxManifestReader_Capabilities", "[api][!hide]")
     REQUIRE(expected == capabilities);
 }
 
+// Validates manifest capabilities IAppxManifestReader::GetCapabilitiesByCapabilityClass
+TEST_CASE("Api_AppxManifestReader_GetCapabilitiesByCapabilityClass", "[api]")
+{
+    std::string package = "StoreSigned_Desktop_x64_MoviesTV.appx";
+    MsixTest::ComPtr<IAppxPackageReader> packageReader;
+    MsixTest::InitializePackageReader(package, &packageReader);
+    MsixTest::ComPtr<IAppxManifestReader> manifestReader;
+    REQUIRE_SUCCEEDED(packageReader->GetManifest(&manifestReader));
+    REQUIRE_NOT_NULL(manifestReader.Get());
+    MsixTest::ComPtr<IAppxManifestReader3> manifestReader3;
+    REQUIRE_SUCCEEDED(manifestReader->QueryInterface(UuidOfImpl<IAppxManifestReader3>::iid, reinterpret_cast<void**>(&manifestReader3)));
+
+    std::vector<std::string> expectedValues = 
+    {
+        "internetClient",
+        "privateNetworkClientServer",
+        "videosLibrary",
+        "removableStorage",
+        "enterpriseDataPolicy",
+        "previewStore",
+    };
+
+    MsixTest::ComPtr<IAppxManifestCapabilitiesEnumerator> capabilities;
+    REQUIRE_SUCCEEDED(manifestReader3->GetCapabilitiesByCapabilityClass(APPX_CAPABILITY_CLASS_ALL, &capabilities));

```

Comment:
>APPX_CAPABILITY_CLASS_ALL [](start = 72, length = 25)

We should make some additional packages so we can target other classes.

Reply:
Or really just manifests, no need for an entire package to test it.

---
In reply to: [305433344](https://github.com/Microsoft/msix-packaging/pull/179#discussion_r305433344) [](ancestors = 305433344)